File: runtests/ocaml_simple_json/simple_json_construct.ml
open Simple_json_require;;
open Simple_json_lexer;;
type ___used_t_head_90xasda
and json = 
| JBool of { value : bool }
| JDict of { value : ((string, json) NameValuePair) list }
| JFlt of { value : float }
| JInt of { value : int }
| JList of { elements : (json) list }
| JNull of unit
| JStr of { value : string }
and ('A, 'B) NameValuePair = {
    name : 'A;
    value : 'B;
}
let mk_NameValuePair (name, value) : ('A, 'B) NameValuePair =
     { name;value }
let mk_JStr (value) : json =
    JStr { value }
let mk_JNull () = JNull ()
let mk_JList (elements) : json =
    JList { elements }
let mk_JInt (value) : json =
    JInt { value }
let mk_JFlt (value) : json =
    JFlt { value }
let mk_JDict (value) : json =
    JDict { value }
let mk_JBool (value) : json =
    JBool { value }

File: runtests/python_simple_json/simple_json_construct.py
from __future__ import annotations as __01asda1ha
from lark import Token as token
import dataclasses as dataclasses
import typing as typing
from .simple_json_require import (appendList,unesc,getStr,parseFlt,parseInt)
@dataclasses.dataclass
class JBool:
    value: bool
@dataclasses.dataclass
class JDict:
    value: list[NameValuePair[str, Json]]
@dataclasses.dataclass
class JFlt:
    value: float
@dataclasses.dataclass
class JInt:
    value: int
@dataclasses.dataclass
class JList:
    elements: list[Json]
@dataclasses.dataclass
class JNull:
    pass
@dataclasses.dataclass
class JStr:
    value: str
if typing.TYPE_CHECKING:
    Json = typing.Union[JStr,JNull,JList,JInt,JFlt,JDict,JBool]
else:
    Json = (JStr,JNull,JList,JInt,JFlt,JDict,JBool)
@dataclasses.dataclass
class NameValuePair:
    name: object
    value: object
NameValuePair = NameValuePair
